File: internal/models/weight.go
package models

import (
	"time"
)

type Weight struct {
	ID           uint      `gorm:"primaryKey"`
	AnimalID     uint      `gorm:"not null"`
	Animal       Animal    `gorm:"foreignKey:AnimalID"`
	Date         time.Time `gorm:"not null"`
	AnimalWeight float64   `gorm:"not null"`
	CreatedAt    time.Time
	UpdatedAt    time.Time
}

File: src/pages/contents/AnimalTable/domain/model/weight.ts
export interface Weight {
  id: number;
  animal_id: number;
  animal_name?: string;
  ear_tag?: number;
  date: string;
  animal_weight: number;
  created_at: string;
  updated_at: string;
}

export interface CreateOrUpdateWeightRequest {
  animal_id: number;
  date: string;
  animal_weight: number;
}

export interface UpdateWeightRequest {
  id: number;
  animal_id: number;
  date: string;
  animal_weight: number;
}

